src/reaction_lammps_mupt/input_parser.py

```
@@ -0,0 +1,102 @@
+            raise ValueError("The 'number_of_monomers' key must map to a non-empty dictionary.")
+
+        if len(inputs["monomers"]) != len(inputs["number_of_monomers"]):
+            raise ValueError(
+                f"Length mismatch: 'monomers' has {len(inputs['monomers'])} entries "
+                f"but 'number_of_monomers' has {len(inputs['number_of_monomers'])} entries. They must match."
+            )
+
+    def _validate_smiles(self, smiles: str) -> None:
+        mol = Chem.MolFromSmiles(smiles)
+        if mol is None:
+            raise ValueError(f"Invalid SMILES string: {smiles!r}. Terminating program.")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)